packages/api/src/task/scheduler.ts

```diff
@@ -298,5 +305,23 @@ export default class TaskScheduler {
         },
       },
     });
+    if (asset.status.phase == "ready" || asset.status.phase == "failed") {
+      let asset_event: EventKey = "asset.ready";
+      if (asset.status.phase === "failed") {
+        asset_event = "asset.failed";
+      }
```

Comment:
Same regarding shortening this! If you get some typing issues I think you can do this:
```suggestion
      const assetEvent = `asset.${asset.status.phases}` as EventKey;
```

Or rather, we don't even need the intermediate variable anymore in either of these, can just declare the routing key:
```ts
let routingKey = `events.asset.${asset.status.phase}` as RoutingKey;
```